ci/check-release-notes.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python3
+
+import argparse
+import yaml
+import re
+import sys
+
+def validate_release_notes_structure(data, file_name):
+    valid = True
+
+    if not isinstance(data, dict):
+        print(f"Error: YAML root should be a dictionary in {file_name}.")
+        valid = False
+
+    if 'releases' not in data:
+        print(f"Error: 'releases' key not found in {file_name}.")
+        valid = False
+
+    if not isinstance(data.get('releases'), dict):
+        print(f"Error: 'releases' key should contain a dictionary in {file_name}.")
+        valid = False
+
+    for release, content in data.get('releases', {}).items():
+        if not re.match(r'^\d+\.\d{8}\.\d+\.\d+$', release):
+            print(f"Error: Release key '{release}' should follow the pattern 'x.yyyymmdd.n.m' in {file_name}.")
+            valid = False
+
+        release_line = f"{release}:"
+        if not release_line.endswith(':'):
+            print(f"Error: Release key '{release}' does not end with a colon ':' in {file_name}.")
+            valid = False
+
+        if not isinstance(content, dict):
+            print(f"Error: Release {release} should contain a dictionary in {file_name}.")
+            valid = False
+
+        if 'issues' not in content:
+            print(f"Error: 'issues' key not found for release {release} in {file_name}.")
+            valid = False
+
+        if not isinstance(content.get('issues'), list):
+            print(f"Error: 'issues' key for release {release} should contain a list in {file_name}.")
+            valid = False
+
+        for issue in content.get('issues', []):
+            if not isinstance(issue, dict):
+                print(f"Error: Each issue for release {release} should be a dictionary in {file_name}.")
+                valid = False
+
+            if 'text' not in issue:
+                print(f"Error: 'text' key not found in issue for release {release} in {file_name}.")
+                valid = False
+            if 'url' not in issue:
+                print(f"Error: 'url' key not found in issue for release {release} in {file_name}.")
+                valid = False
+
+    return valid
+
+def validate_indentation(file_path):
```

Comment:
Hmm, unsure about this check. The actual indentation used isn't really important unless it affects the actual YAML document. Either it won't parse correctly (caught earlier), or it will change the object definition (caught by `validate_release_notes_structure()`).

Reply:
Yes, I have removed this function